### DISCUSSION TITLE
Welche Batteriesteuerung anwenden?

### DISCUSSION THREAD

**Discussion by Obsthaendler** [bolliy/ioBroker.sun2000#137] [Allgemein]
Hallo,

ich versuche derzeit durch diverse Skripte eine optimale Batterielogik zu entwerfen, die in Verbindung mit einem dynamischen Stromtarif die Batterie zur richtigen Zeit lädt und auch bei anderen Dingen sperrt. Im Forum werden einige Beispiele genannt.

Innerhalb des Wikis hier werden verschiedene Steuerungen erklärt. 
https://github.com/bolliy/ioBroker.sun2000/wiki/Battery-control oder https://github.com/bolliy/ioBroker.sun2000/wiki/Erzwungenes-Laden-und-Entladen-der-Batterie-(Force-charge-discharge-battery)

Ich hatte nun versucht durch den Artikel "Laden des Speichers über dynamische Stromtarife" den Ziel-SOC zu setzen. Leider passiert hierbei das die Batterie in den Standby geht, weil die AC-Steuerung deaktiviert wird wenn Ziel SOC erreicht wird und die Batterie zu schnell leer ist.

Welche Möglichkeiten nutzt ihr? Was sind die Unterschiede von:

sun2000.0.inverter.x.control.battery.chargingCutoffCapacity
sun2000.0.inverter.x.control.battery.targetSOC
sun2000.0.inverter.x.control.battery.gridChargeCutoffSOC

Viele Grüße
Steffen

**Comment by bolliy** [comment]
Hallo Steffen,
im Forum gab es schon das Thema: https://forum.iobroker.net/post/1211778
Ich werde noch versuchen eine Lösung für das Aufwachen des Speichers zu finden....

LG Stephan

**Reply by Obsthaendler** [reply]
Das die Batterie wach bleibt sind aber eigentlich andere Dinge die damit zu tun haben. Ich hatte nämlich nun auch den Fall das AC-seitiges laden aus war, durch das setzen von AC TargetSOC, somit ist der Speicher Offline gegangen, weil das Ziel erreicht war. Anschließend reichte die Restkapazität nicht mehr vollständig bis Sonne den Speicher wieder auflädt. Somit hatte der Speicher 0%. Ergo, der WR Master ging auch in Standby. Parametereinstellungen dann zu verändern sind nicht möglich. 

Entweder man lässt AC-laden dauerhaft an, sodass die Aufrechterhaltung der Batterie gegeben ist oder man muss sich eine gute Skriptlogik überlegen, um den entgegen zu steuern. 

Ich möchte hier in der Diskussion auch nur gerne die Eigenschaften diskutieren, was beim Nutzen der genannten Control States so im System passiert und dann das beste UseCase für mich erstellen.

**Comment by bolliy** [comment]
Ich habe heute abend im standby meines WR einiges probiert. Sofern ich **AC-laden** aktiviere, dauert es ca. 20 Minuten bis der sun2000 + der Speicher aufwachen. Dann kann man über Battery-control (wirgingMode = 5) oder mit erzwungenem Laden den Speicher aus dem Netz füllen. Erst wenn der Vorgang abgeschlossen ist erfolgt ein Deaktivieren von AC-laden. Dann brauch der WR ca. 1 Minute um wieder in den standby zu gehen. 
Wenn man also in seiner Logik 30 Minuten vor dem eigentlichen Laden den WR aufweckt sollte dein Vorhaben umsetzbar sein ;)

Ich werde das Wiki bei zeiten mal anpassen ....
